homeassistant/components/trafikverket_train/coordinator.py

```diff
@@ -0,0 +1,147 @@
+"""DataUpdateCoordinator for the Trafikverket Train integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import date, datetime, time, timedelta
+import logging
+
+from pytrafikverket import TrafikverketTrain
+from pytrafikverket.trafikverket_train import StationInfo, TrainStop
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_WEEKDAY, WEEKDAYS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+from homeassistant.util import dt as dt_util
+
+from .const import CONF_TIME, DOMAIN
+
+
+@dataclass
+class TrainData:
+    """Dataclass for Trafikverket Train data."""
+
+    departure_time: datetime | None
+    departure_state: str
+    cancelled: bool
+    delayed_time: int | None
+    planned_time: datetime | None
+    estimated_time: datetime | None
+    actual_time: datetime | None
+    other_info: str | None
+    deviation: str | None
+
+
+_LOGGER = logging.getLogger(__name__)
+TIME_BETWEEN_UPDATES = timedelta(minutes=5)
+
+
+def next_weekday(fromdate: date, weekday: int) -> date:
```

Comment:
It seems this is only used in this module?
If so, make it protected:
```suggestion
def _next_weekday(fromdate: date, weekday: int) -> date:
```